lib/spack/spack/directives.py

```diff
@@ -455,8 +463,14 @@ def _execute_version(pkg, ver, **kwargs):
     pkg.versions[version] = kwargs
 
 
-def _depends_on(pkg, spec, when=None, type=dt.DEFAULT_TYPES, patches=None):
-    when_spec = make_when_spec(when)
+def _depends_on(
+    pkg: "spack.package_base.PackageBase",
+    spec: SpecType,
+    when: WhenType = None,
+    type: DepType = dt.DEFAULT_TYPES,
+    patches: PatchesType = None,
+):
```

Comment:
(minor) Asking for explicit keyword arguments usually makes the calling site more legible:
```suggestion
def _depends_on(
    pkg: "spack.package_base.PackageBase",
    *,
    spec: SpecType,
    when: WhenType = None,
    type: DepType = dt.DEFAULT_TYPES,
    patches: PatchesType = None,
):
```

Reply:
I agree -- however I think `spec` should be positional. Did you mean to include it?

Reply:
Yep, apologies, it's a typo

Reply:
fixed